build_tools/waf_dynamo.py

```diff
@@ -1537,7 +1537,7 @@ def detect(conf):
         conf.env['STATICLIB_MARKER']=''
         conf.env['SHLIB_MARKER']=''
 
-    if platform in ('x86_64-linux',): # Currently the only platform exhibiting the behavior
+    if platform in ('x86_64-linux','armv7-android','arm64-android'): # Currently the only platform exhibiting the behavior
```

Comment:
The issue was about ordering.
Different platforms want the libs specified in different orders.